app/src/main/java/org/stepic/droid/notifications/badges/NotificationsBadgesManager.kt

```diff
@@ -0,0 +1,80 @@
+package org.stepic.droid.notifications.badges
+
+import android.content.Context
+import android.support.annotation.MainThread
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig
+import io.reactivex.Scheduler
+import io.reactivex.Single
+import me.leolin.shortcutbadger.ShortcutBadger
+import org.stepic.droid.base.ListenerContainer
+import org.stepic.droid.configuration.RemoteConfig
+import org.stepic.droid.di.AppSingleton
+import org.stepic.droid.di.qualifiers.BackgroundScheduler
+import org.stepic.droid.di.qualifiers.MainScheduler
+import org.stepic.droid.notifications.model.NotificationStatuses
+import org.stepic.droid.preferences.SharedPreferenceHelper
+import org.stepic.droid.web.Api
+import javax.inject.Inject
+
+@AppSingleton
+class NotificationsBadgesManager
+@Inject
+constructor(
+        private val api: Api,
+        private val context: Context,
+        private val sharedPreferenceHelper: SharedPreferenceHelper,
+        private val firebaseRemoteConfig: FirebaseRemoteConfig,
+        private val listenerContainer: ListenerContainer<NotificationsBadgesListener>,
+
+        @BackgroundScheduler
+        private val scheduler: Scheduler,
+        @MainScheduler
+        private val mainScheduler: Scheduler
+) {
+    fun fetchAndThenSyncCounter() {
```

Comment:
`fetchAndThenSyncCounter` is not correct name, because fetching and `syncCounter` are executed in parallel. And it can execute `syncCounter` and after that execute `updateCounter`.

So, `fetch` is strange too. Because this method updateAndSyncCounter. The order is not determined, is it bug?